init_process.go

```diff
@@ -0,0 +1,406 @@
+func createS3ConfigFileFolder(folderPath string) {
+	os.Mkdir(folderPath, os.ModePerm)
+}
+
+// get our built in config file or files from s3
+// process built-in config files directly
+// add S3 config files to folder "init_process_fluent-bit_s3_config_files"
+func getAllConfigFiles() {
+
+	//get all env vars
+	envs := os.Environ()
+	//find all env vars match specified prefix
+	for _, env := range envs {
+		var envKey string
+		var envValue string
+		env_kv := strings.SplitN(env, "=", 2)
+		if len(env_kv) != 2 {
+			log.Printf("Unrecognizable environment variables: %s\n", env)
+			continue
+		}
+
+		envKey = string(env_kv[0])
+		envValue = string(env_kv[1])
+
+		matched_s3, _ := regexp.MatchString("aws_fluent_bit_s3_", envKey)
+		matched_file, _ := regexp.MatchString("aws_fluent_bit_file_", envKey)
+		//if this env var's value is an arn
+		if matched_s3 {
+			getS3ConfigFile(envValue)
+		}
+		//if this env var's value is a path of our built-in config file
+		if matched_file {
+			processBuiltInConfigFile(envValue)
+		}
+	}
+}
+
+// process built-in config files
+func processBuiltInConfigFile(path string) {
+	content_b, err := ioutil.ReadFile(path)
+	if err != nil {
+		log.Println(err)
+		log.Fatalf("Cannot open file: %s\n", path)
+	}
+	content := string(content_b)
+	if strings.Contains(content, "[PARSER]") {
+		// this is a parser config file, change command
+		updateCommand(path)
+	} else {
+		// this is not a parser config file. @INCLUDE
+		writeInclude(path, MainConfigFilePath)
+	}
+}
+
+// download S3 config file to S3 config file folder
+func getS3ConfigFile(arn string) {
+	if !exists3Client {
+		createS3Client()
+	}
+
+	//e.g. "arn:aws:s3:::ygloa-bucket/s3_parser.conf"
+	arnBucketFile := arn[13:]
+	bucketAndFile := strings.SplitN(arnBucketFile, "/", 2)
+	if len(bucketAndFile) != 2 {
+		log.Fatalf("Unrecognizable arn: %s\n", arn)
+	}
+
+	bucketName := bucketAndFile[0]
+	s3FilePath := bucketAndFile[1]
+
+	// get bucket region
+	input := &s3.GetBucketLocationInput{
+		Bucket: aws.String(bucketName),
+	}
+
+	output, err := s3Client.GetBucketLocation(input)
+	if err != nil {
+		log.Println(bucketName + ":" + s3FilePath)
+		log.Println(err)
+		log.Println("Cannot get bucket region")
+	}
+
+	bucketRegion := aws.StringValue(output.LocationConstraint)
+	// Buckets in Region us-east-1 have a LocationConstraint of null.
+	// https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#GetBucketLocationOutput
+	if bucketRegion == "" {
+		bucketRegion = "us-east-1"
+	}
+
+	// create downloader
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(bucketRegion)},
+	)
+	if err != nil {
+		log.Println(err)
+		log.Fatalln("Cannot creat a new session")
+	}
+
+	// need to specify session region!
+	s3Downloader := s3manager.NewDownloader(sess)
+
+	// download file and store
+	s3FileName := strings.SplitN(s3FilePath, "/", -1)
+	fileFromS3, err := os.Create(ConfigFileFolderPath + "/" + s3FileName[len(s3FileName)-1])
```

Comment:
Is it the same as `createfile` function?

Reply:
Yeah, already modified